platforms/platforms.go

```diff
@@ -43,10 +44,29 @@ func DetermineBazelFilename(version string, includeSuffix bool) (string, error)
 		return "", fmt.Errorf("unsupported operating system \"%s\", must be Linux, macOS or Windows", runtime.GOOS)
 	}
 
+	if osName == "darwin" {
+		machineName = DarwinFallback(machineName, version)
+	}
+
 	var filenameSuffix string
 	if includeSuffix {
 		filenameSuffix = DetermineExecutableFilenameSuffix()
 	}
 
 	return fmt.Sprintf("bazel-%s-%s-%s%s", version, osName, machineName, filenameSuffix), nil
 }
+
+// DarwinFallback Darwin arm64 was supported since 4.1.0, before 4.1.0, fall back to x86_64
+func DarwinFallback(machineName string, version string) (alterMachineName string) {
+	v, err := sem_ver.NewVersion(version)
+	if err != nil {
+		return machineName
+	}
+
+	armSupportVer, _ := sem_ver.NewVersion("4.1.0")
+
+	if machineName == "arm64" && v.LessThan(armSupportVer) {
+		return "x86_64"
```

Comment:
Please log a warning so that users aren't too surprised.

Reply:
fixed